frontend_app_share_folder_advanced/src/container/ShareFolderAdvanced.jsx

```diff
@@ -1,290 +1,280 @@
+  }
+
+  sendGlobalFlashMessage = (msg, type = 'info') => GLOBAL_dispatchEvent({
+    type: CUSTOM_EVENT.ADD_FLASH_MSG,
+    data: {
+      msg: msg,
+      type: type,
+      delay: undefined
+    }
+  })
+
+  loadContent = async () => {
+    const { props, state } = this
+
+    const fetchContentTypeList = await handleFetchResult(await getContentTypeList(state.config.apiUrl))
+
+    switch (fetchContentTypeList.apiResponse.status) {
+      case 200: this.setState({ tracimContentTypeList: fetchContentTypeList.body.filter(ct => ct.slug !== 'comment') }); break
+      default: this.sendGlobalFlashMessage(props.t("Error while loading tracim's content type list"), 'warning')
+    }
+  }
+
+  loadImportAuthorizationsList = async () => {
+    const { content, config } = this.state
+
+    const fetchResultImportAuthorizationsList = await handleFetchResult(await getImportAuthorizationsList(config.apiUrl, content.workspace_id))
+
+    switch (fetchResultImportAuthorizationsList.apiResponse.status) {
+      case 200:
+        this.setState({
+          uploadLinkList: fetchResultImportAuthorizationsList.body
+        })
+        GLOBAL_dispatchEvent({ type: CUSTOM_EVENT.REFRESH_CONTENT_LIST, data: {} })
+        break
+      default:
+        this.sendGlobalFlashMessage(this.props.t('Error while loading share links list'))
+    }
+  }
+
+  handleChangeEmails = e => this.setState({ uploadEmails: e.target.value })
+  handleChangePassword = e => this.setState({ uploadPassword: e.target.value })
+
+  handleClickBtnCloseApp = () => {
+    this.setState({ isVisible: false })
+    GLOBAL_dispatchEvent({ type: CUSTOM_EVENT.APP_CLOSED, data: {} })
+  }
+
+  handleClickDeleteImportAuthorization = async importAuthorizationId => {
+    const { config, content } = this.state
+    const { props } = this
+
+    const fetchResultDeleteImportAuthorization = await handleFetchResult(
+      await deleteImportAuthorization(config.apiUrl, content.workspace_id, importAuthorizationId)
+    )
+
+    switch (fetchResultDeleteImportAuthorization.status) {
+      case 204:
+        this.loadImportAuthorizationsList()
+        break
+      case 400:
+        this.sendGlobalFlashMessage(props.t('Error in the URL'))
+        break
+      default: this.sendGlobalFlashMessage(props.t('Error while deleting share link'))
+    }
+  }
+
+  handleClickNewUploadComponent = () => {
+    this.setState({ currentPage: this.UPLOAD_STATUS.NEW_UPLOAD })
+  }
+
+  handleClickNewUpload = async () => {
+    const { state, props } = this
+
+    let uploadEmailList = parserStringToList(state.uploadEmails)
+    let invalidEmails = []
+
+    uploadEmailList.forEach(uploadEmail => {
+      if (!checkEmailValidity(uploadEmail)) invalidEmails.push(uploadEmail)
+    })
+
+    uploadEmailList = uploadEmailList.filter(uploadEmail => !invalidEmails.includes(uploadEmail))
+
+    if (invalidEmails.length > 0) {
+      this.sendGlobalFlashMessage(props.t(`Error: ${invalidEmails} are not valid`))
+    }
```

Comment:
should we still upload the file if an email is invalid ?

Reply:
done